spec/mongoid/association/depending_spec.rb

```diff
@@ -865,5 +865,21 @@ class DoubleAssocThree < DoubleAssoc; end
         expect(person.delete).to be(true)
       end
     end
+
+    context 'when deleted inside a transaction' do
+      require_topology :replica_set
+      
+      before do
+        person.restrictable_posts << post
+      end
+
+      it 'doesn\'t raise an exception' do
+        person.with_session do |session|
+          session.start_transaction
+          person.destroy
+          session.commit_transaction
+        end
+      end
+    end
```

Comment:
Could you add the following tests:
1. Test that this works with `has_and_belongs_to_many`
2. Add unit tests for the `any?` method
3. Add unit tests for the `any?` method on the `has_and_belongs_to_many` relation?